lightning/src/ln/chan_utils.rs

```diff
@@ -822,88 +870,100 @@ impl CommitmentTransaction {
+	pub(crate) fn new_with_auxiliary_htlc_data<T>(commitment_number: u64, to_broadcaster_value_sat: u64, to_countersignatory_value_sat: u64, keys: TxCreationKeys, feerate_per_kw: u32, htlcs: Vec<HTLCOutputInCommitment>, aux: Vec<T>, channel_parameters: &DirectedChannelTransactionParameters) -> (CommitmentTransaction, Vec<(HTLCOutputInCommitment, T)>) {
 		// Sort outputs and populate output indices while keeping track of the auxiliary data
-		let mut txouts = Self::internal_build_outputs(&keys, to_broadcaster_value_sat, to_countersignatory_value_sat, &htlcs, aux, channel_parameters, &secp_ctx).unwrap();
+		let mut txouts = Self::internal_build_outputs(&keys, to_broadcaster_value_sat, to_countersignatory_value_sat, &htlcs, aux, channel_parameters).unwrap();
 		let mut result_htlcs_with_aux = Vec::new();
 		let mut htlcs = Vec::new();
+		let mut outputs = Vec::new();
 		for (idx, mut out) in txouts.drain(..).enumerate() {
 			if let Some(mut htlc) = (out.1).1.take() {
 				htlc.0.transaction_output_index = Some(idx as u32);
 				result_htlcs_with_aux.push((htlc.0.clone(), htlc.1));
 				htlcs.push(htlc.0);
 			}
+			outputs.push(out.0);
 		}
 
+		let (obscured_commitment_transaction_number, txins) = Self::internal_build_inputs(commitment_number, channel_parameters);
+		let transaction = Self::make_transaction(obscured_commitment_transaction_number, txins, outputs);
+		let txid = transaction.txid();
 		let info = CommitmentTransaction {
 			commitment_number,
 			to_broadcaster_value_sat,
 			to_countersignatory_value_sat,
 			feerate_per_kw,
 			htlcs,
 			keys,
+			built: BuiltCommitmentTransaction {
+				transaction,
+				txid
+			},
 		};
 		(info, result_htlcs_with_aux)
 	}
 
-	pub(crate) fn calculate_txid<T: secp256k1::Signing + secp256k1::Verification>(&self, channel_parameters: &DirectedChannelTransactionParameters, secp_ctx: &Secp256k1<T>) -> Txid {
-		let tx = self.build(channel_parameters, secp_ctx).unwrap().0;
-		tx.txid()
+	pub(crate) fn trust_txid(&self) -> Txid {
+		self.built.txid
 	}
 
-	/// Build the Bitcoin transaction
-	pub fn build<T: secp256k1::Signing + secp256k1::Verification>(&self, channel_parameters: &DirectedChannelTransactionParameters, secp_ctx: &Secp256k1<T>) -> Result<(bitcoin::Transaction, Vec<Script>), ()> {
-		let (obscured_commitment_transaction_number, txins) = self.build_inputs(channel_parameters);
+	/// Trust the pre-built commitment transaction.
+	///
+	/// This should only be used if you fully trust the builder of this object.  It should not
+	/// be used by an external signer - instead use the build function.
+	pub fn trust_built_transaction(&self) -> &BuiltCommitmentTransaction {
```

Comment:
A _method_ name starting with a verb indicates an action performed on the object, not an action taken by the caller. That's why starting the method name with "trust" seems strange to me. I'd imagine naming this method `untrusted_built_transaction` would better describe -- "untrusted" being an adjective  -- what is returned from the accessor method.

Reply:
That sounds reasonable.  @TheBlueMatt - since this is a change to the convention we previously settled on (in the context of keys), what do you think?

Reply:
Sure, either naming convention is fine with me, Jeff's take seems reasonable.